### PR TITLE
Improve sync

### PR DESCRIPTION
**Pull request by xiormeesh**
- sync performed for specific building (passed to command or in url)
- now synchronizing only pointers belonging to category "person"
- improved error reporting

### CODE REVIEW

File: app/Helpers/RemoteData.php [hunk 1]
```diff
@@ -43,42 +45,47 @@ public static function sync()
...
             "successful" => 0,
             "failed"=> 0
         );
 
-        # TODO: get only pointers with category "Person"
-        Pointer::all()->each(function (Pointer $pointer) use ($users, &$result) {
+        foreach ($building->levels as $level) {
 
-            $filtered = $users->filter(function($value) use ($pointer){
+            foreach (Level::find($level->id)->pointers as $pointer) {
```
Comment (othercodes) [inline app/Helpers/RemoteData.php]: why not directly iterate: 
```php
foreach($level->pointers as $pointer) {
   // process pointer
}
```

File: app/Helpers/RemoteData.php [hunk 2]
```diff
@@ -43,42 +45,47 @@ public static function sync()
...
 
-                $result['successful']++;
+                        $pointer->name = "{$match['givenName']} {$match['surname']}";
+                        $pointer->description = "Job Title: {$match['jobTitle']} <br> Department: {$match['department']}";
+                        $pointer->save();
 
-            } else {
+                        $result['successful']++;
 
-                Log::critical("Couldn't find user with email: {$pointer->meta}.");
```
Comment (othercodes) [inline app/Helpers/RemoteData.php]: never use critical!!! use log::error instead please.

File: app/Http/Controllers/AuthController.php [hunk 3]
```diff
@@ -66,7 +66,7 @@ public function callback(Request $request) {
                 $token->storeTokens($accessToken, $user);
 
                 # once auth is done and token successfully saved in db, we can repeatedly sync through /sync
-                return redirect('/app/sync');
+                return redirect('/1/sync');
```
Comment (othercodes) [inline app/Http/Controllers/AuthController.php]: Harcoded :( ?

File: app/Helpers/RemoteData.php (hunk 1 above)
Comment (othercodes) [reply app/Helpers/RemoteData.php]: in this case, we can save one db call, the pointers are eagerly loaded, so we can directly use them.

File: app/Helpers/RemoteData.php (hunk 2 above)
Comment (othercodes) [reply app/Helpers/RemoteData.php]: Critical level should be used in more extreme or critical conditions, in this case the app can keep working normally, this error is raised for 1 item that not compromise the functional integrity of the app so imho this should be error or even warning level.

Check https://tools.ietf.org/html/rfc5424#page-11.

File: app/Helpers/RemoteData.php (hunk 1 above)
Comment (xiormeesh) [reply app/Helpers/RemoteData.php]: agree, missed that for some reason

File: app/Helpers/RemoteData.php (hunk 2 above)
Comment (xiormeesh) [reply app/Helpers/RemoteData.php]: fixed

**Comment by xiormeesh** [comment]
- now authentication data is linked to a building
- new behavior for views /signin, /callback, /sync
    - if no id is passed fetching the first auth data from db (/signin -> /signin/1)
    - if building with the id is not found, throwing NotFoundHttpException (signin/15)

**Pull request merged**